confmap/provider/s3provider/provider.go

```diff
@@ -50,39 +54,41 @@ type provider struct {
 //
 // Examples:
 // `s3://DOC-EXAMPLE-BUCKET.s3.us-west-2.amazonaws.com/photos/puppy.jpg` - (unix, windows)
-func New() confmap.Provider {
-	return &provider{client: nil}
+func New(optFns ...func(*config.LoadOptions) error) confmap.Provider {
```

Comment:
At this time we do not have any config providers that accept parameters on initialization.  We also don't have a formalized method for including new config providers in the collector builder and I worry that taking arbitrary parameters in the initializer could complicate doing so.  

@neelayu can you please provide some more context on why this is required and the existing environment-based configuration of the underlying AWS SDK is insufficient?

@open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers I'd like your feedback on this.

Reply:
Your point is valid. Although the current state of the collector builder(ocb) has no support for custom providers as far as I know.(of course that can change). This would generally be used by distributions not using the builder.

@bryan-aguilar suggested that there are other ways of configuring the creds source. I thought of extending that support in the provider.